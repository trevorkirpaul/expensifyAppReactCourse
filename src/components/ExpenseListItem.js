import React from 'react';

import { removeExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, note, createdAt, id, dispatch }) => (
  <div>
    <h3>      
      <Link to={`/edit/${id}`}>{description}</Link>
      
    </h3>
    <p>Amount: {amount}</p>

    {createdAt !== 0 && <p>Created: {createdAt}</p>}    

    {note !== '' && <p>Note: {note}</p> }
    
  </div>
);

export default ExpenseListItem;