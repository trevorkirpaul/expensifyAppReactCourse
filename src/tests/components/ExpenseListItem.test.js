import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';



test('should render ExpenseListItem', () => {
  const wrapper = shallow(<ExpenseListItem key={1} { ...expenses[0] } />);
  expect(wrapper).toMatchSnapshot();
});