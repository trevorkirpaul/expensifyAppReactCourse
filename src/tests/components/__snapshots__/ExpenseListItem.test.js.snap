// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      Gum
    </Link>
  </h3>
  <p>
    Amount: 
    195
  </p>
</div>
`;
